### PR TITLE
Add debugging guide to docs

### PR DESCRIPTION
**Pull request by dnth**
## Describe changes
This PR adds debugging guide to our docs page.

I added the following items:
- How to post on Slack (they should tell their stack, python env, code etc)
- How to search Slack for answers (Maybe using [https://open.crowd.dev/zenml](https://open.crowd.dev/zenml))
- How to see logs on the client and server
- how to toggle ZENML_LOGGING_VERBOSITY env variable to show more logs
- Error FAQ.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by htahir1 — COMMENTED** [review]
I like where this is going! I have some suggestions to add:

* When asking on slack, please should always post their full error traceback. If the full error does not fit they can attach a file or use a service like pastebin (https://pastebin.com/). 
* Along with their traceback, they should also always give the logs of the orchestrator if relevant. For example, if using the kubeflow orchestrator they should give the logs of the pod that is running the step that failed.
* They should also always share the output of the following commands: `zenml status` and `zenml stack describe` 

Apart from these, iI think we need a bit of a guide for the top level headings. For exampe, when search slack for answers? When should one see logs of client and server ? When you should one toggle verbosity? The story should be clear as to what a user should do when encountering an error

File: docs/book/misc/debugging.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+This page documents a series of how-tos for efficient debugging and getting help without losing your hair.
+
+## 💡 Introduction
+If you stumbled upon this page, chances are you're facing issues with using ZenML. First, thank you for your interest in ZenML!
+We want to make your journey with ZenML as easy and smooth as possible when faced with issues.
+This page documents suggestions and best practices to let you debug, get help, and solve issues quickly.
+
+## 🔍 Getting help
+The most direct and quickest path to get help is on our Slack. 
+Join our Slack channel [here](https://zenml.io/slack-invte).
```
Comment (htahir1) [inline docs/book/misc/debugging.md:15]: Typo . /slack-invite. Btw from now on we also have /slack as another link to the same

**Review by htahir1 — CHANGES_REQUESTED** [review]

File: docs/book/misc/debugging.md (hunk 1 above)
Comment (dnth) [reply docs/book/misc/debugging.md]: Fixed!

File: docs/book/misc/debugging.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
+---
+description: A guide to debug common issues and get help.
+---
+
+# Debugging Guide
+This page documents a series of how-tos for efficient debugging and getting help without losing your hair.
```
Comment (htahir1) [inline docs/book/misc/debugging.md:6]: `without losing your hair` maybe isnt the right tone. How about `efficient debugging and getting help.`

File: docs/book/misc/debugging.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+## 💡 Introduction
+If you stumbled upon this page, chances are you're facing issues with using ZenML. First, thank you for your interest in ZenML!
+We want to make your journey with ZenML as easy and smooth as possible when faced with issues.
+This page documents suggestions and best practices to let you debug, get help, and solve issues quickly.
```
Comment (htahir1) [inline docs/book/misc/debugging.md:11]: I wouldnt really make it a big deal, its expected that we help them

```suggestion
If you stumbled upon this page, chances are you're facing issues with using ZenML.
This page documents suggestions and best practices to let you debug, get help, and solve issues quickly.
```

File: docs/book/misc/debugging.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+## 🔍 Getting help
+The most direct and quickest path to get help is on our Slack. 
+Join our Slack channel [here](https://zenml.io/slack).
+
+### 🔷 When to get help?
```
Comment (htahir1) [inline docs/book/misc/debugging.md:17]: I feel like this should be a top level section that is at the start and not nested under Getting Help

File: docs/book/misc/debugging.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+# Debugging Guide
+This page documents a series of how-tos for efficient debugging and getting help without losing your hair.
+
+## 💡 Introduction
+If you stumbled upon this page, chances are you're facing issues with using ZenML. First, thank you for your interest in ZenML!
+We want to make your journey with ZenML as easy and smooth as possible when faced with issues.
+This page documents suggestions and best practices to let you debug, get help, and solve issues quickly.
+
+## 🔍 Getting help
+The most direct and quickest path to get help is on our Slack.
```
Comment (htahir1) [inline docs/book/misc/debugging.md:14]: I would not encourage people to ask for help on slack. First I would guide them to get additional logs, get server and client logs, and try to understand the problem. Only thne shift them to slack

**Review by htahir1 — CHANGES_REQUESTED** [review]
Greeat! Left some more feedback

File: docs/book/misc/debugging.md (hunk 2 above)
Comment (dnth) [reply docs/book/misc/debugging.md]: Fixed.

File: docs/book/misc/debugging.md (hunk 5 above)
Comment (dnth) [reply docs/book/misc/debugging.md]: Fixed.

File: docs/book/misc/debugging.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+
+When applicable, also attach logs of the orchestrator. For example, if you're using the Kubeflow orchestrator, include the logs of the pod that was running the step that failed.
+
+Usually, the default log you see in your terminal is sufficient, in the event it's not, then it's useful to provide additional logs.
+Additional logs are not shown by default, you'll have to toggle an environment variable for it.
+Read the next section to find out how.
+
+
+## 🔄 Additional logs
+When the default logs are not helpful, ambiguous, or does not point you to the root of the issue, you can toggle the value of the `ZENML_LOGGING_VERBOSITY` environment variable to change the type of logs shown.
```
Comment (htahir1) [inline docs/book/misc/debugging.md:80]:
```suggestion
When the default logs are not helpful, ambiguous, or do not point you to the root of the issue, you can toggle the value of the `ZENML_LOGGING_VERBOSITY` environment variable to change the type of logs shown.
```

File: docs/book/misc/debugging.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+You can pick other values such as `WARN`, `ERROR`, `CRITICAL`, `DEBUG` to change what's shown in the logs.
+And export the environment variable in your Linux/macOS terminal. 
+For example:
+```
+export ZENML_LOGGING_VERBOSITY=DEBUG
+```
+
+Read more about how to set environment variables [here](https://linuxize.com/post/how-to-set-and-list-environment-variables-in-linux/).
```
Comment (htahir1) [inline docs/book/misc/debugging.md:94]: This just points to Linux, we should cater for all OS's

**Review by htahir1 — CHANGES_REQUESTED** [review]
A few comments but overall almost there!

**Review by htahir1 — APPROVED** [review]
LGTM!

**Pull request merged**